src/Core/include/OpenLoco/Core/EnumFlags.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+
+namespace OpenLoco::Core
+{
+
+#define OPENLOCO_ENABLE_ENUM_OPERATORS(ENM)                                                                                 \
+    static_assert(std::is_unsigned_v<std::underlying_type_t<ENM>> == true, "Underlying enum type must be unsignd");         \
```

Comment:
Do you think an additional assert that checks that `ENM::none` exists would be good? Then we have a little bit of consistency.

Reply:
Its possible to do, but prior to C++20 not really an easy way without adding a bunch of template code. I can just add a static_assert but in case that member doesn't exist it won't be the static_assert message rather a compiler error.

Reply:
Compiler error will be fine. I think I've merged something that uses these flags so will need a rebase.

Reply:
Added static_assert